workshops/urls.py

```diff
@@ -20,7 +20,12 @@
         url(r'^membership/add/$', views.membership_create, name='membership_add'),
     ])),
 
+    url(r'^memberships/', include([
+        url(r'^$', views.all_memberships, name='all_memberships'),
+        url(r'^add/$', views.MembershipCreate.as_view(), name='membership_add'),
+    ])),
```

Comment:
I'd try to use the CBV `MembershipCreate` and change the earlier url `^organization/(?P<org_domain>[\w\.-]+)/membership/add/$` to `^memberships/add/(?P<org_domain>[\w\.-]+)/$`.